integration-cli/docker_cli_run_test.go

```diff
@@ -2826,3 +2826,48 @@ func (s *DockerSuite) TestRunCreateContainerFailedCleanUp(c *check.C) {
 	containerID, err := inspectField(name, "Id")
 	c.Assert(containerID, check.Equals, "", check.Commentf("Expected not to have this container: %s!", containerID))
 }
+
+// TestRunNonExecutableCmd checks that 'docker run busybox foo' exits with error code 127'
+func (s *DockerSuite) TestRunNonExecutableCmd(c *check.C) {
+	runCmd := exec.Command(dockerBinary, "run", "busybox", "foo")
+	out, exit, err := runCommandWithOutput(runCmd)
+	if !(err != nil && exit == 127) {
+		c.Fatalf("Run non-executable command should have errored with exit code 127, but we got out: %s, exit: %d, err: %s", out, exit, err)
+	}
```

Comment:
Can you please add a check of the "docker inspect" output to make sure that State.ExitCode is also 127?  Just to be sure.